### DISCUSSION TITLE
Berkeley QA Initial Release 144feaa

### DISCUSSION THREAD

**Discussion by lk86** [MinaProtocol/mina#10300] [Berkeley]
Mina’s ecosystem partner, O(1) Labs, is continuously working on improvements to the core protocol and today we are excited to announce a new unstable release `144feaa` that is ready to test on the brand new Berkeley QANet! The first comment below describes some concrete steps for what needs to be tested and how you can help, just be sure to report any issues in the comments on this post or tag any issues you create with `berkeley-qa` and include berkeley 144feaa in the issue title.

### New Features/Improvements:
1. New transaction model to support snapps:
New Transaction model RFC https://github.com/MinaProtocol/mina/pull/8068
Updating the transaction model to support solidity features https://github.com/MinaProtocol/mina/pull/9123
[Reference implementation docs](https://o1-labs.github.io/snapps-txns-reference-impl/target/doc/snapps_txn_reference_impl/index.html) (WIP)

2. New hash function for snapp transactions:
Payments and delegations should still use the old format.
https://github.com/MinaProtocol/mina/pull/10005

3. GraphQL changes for snapp transactions
* Added new fields to the `account` query
* Added snapp transactions to `block`, `bestTip` queries
* Added a new query for pooled snapp transactions
* Added a new mutation`sendSnapp` to send snapp transactions

4. Environment variables renamed:
* CODA_CONFIG... to MINA_CONFIG...
* CODA_TIME_OFFSET to `MINA_TIME_OFFSET`.
* `CODA_PRIVKEY_PASS` to `MINA_PRIVKEY_PASS`
* `CODA_LIBP2P_PASS` to `MINA_LIBP2P_PASS`.
* `CODA_KEYS_PATH` to `MINA_KEYS_PATH`
* `CODA_CLIENT_TRUSTLIST` to `MINA_CLIENT_TRUSTLIST`
PRs
https://github.com/MinaProtocol/mina/pull/9115
https://github.com/MinaProtocol/mina/pull/9117
https://github.com/MinaProtocol/mina/pull/9116
https://github.com/MinaProtocol/mina/pull/9114
https://github.com/MinaProtocol/mina/pull/9120

5. Update to the new kimchi proof system
[Kimchi Blog Post](https://minaprotocol.com/blog/kimchi-the-latest-update-to-minas-proof-system)
https://github.com/MinaProtocol/mina/pull/9547
https://github.com/MinaProtocol/mina/pull/9780

6. Proof system upgraded to use the latest arkworks
https://github.com/MinaProtocol/mina/pull/9145

7. Transactions are ordered by fee per weight unit
https://github.com/MinaProtocol/mina/pull/9440

### Known Issues:
- Payments from client sdk are failing with `Invalid Signature` https://github.com/MinaProtocol/mina/issues/10298

### Complete Changelog since 1.3.0:
PR list discussion: https://github.com/MinaProtocol/mina/discussions/10292
Github Compare: https://github.com/MinaProtocol/mina/compare/release/1.3.0...release/2.0.0

### Debian Packages:
**Debian Repository:**
First, set up and update the `unstable` Debian Repository for your platform. Replace the word `CODENAME` with the appropriate codename for your machine, one of `bionic`, `focal`, `stretch`, `buster`, or `bullseye` and run:
```
echo "deb [trusted=yes] http://packages.o1test.net CODENAME unstable" | sudo tee /etc/apt/sources.list.d/mina-unstable.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-berkeley=1.3.0beta1-release-2.0.0-144feaa`
**New Snapps Test Transaction CLI:** `sudo apt-get install -y mina-snapp-test-transaction=1.3.0beta1-release-2.0.0-144feaa`
[ Snapps Transaction CLI Documentation](https://github.com/MinaProtocol/mina/blob/feature/snapp-tool-doc/src/app/snapp_test_transaction/README.md)

### Docker Images:
Every docker image is built for each of the supported platforms, replace `CODENAME` in the images below with your choice of `bionic`, `focal`, `stretch`, `buster`, or `bullseye` to select the base docker image.
**Mina Daemon:** `minaprotocol/mina-daemon:1.3.0beta1-release-2.0.0-144feaa-CODENAME-berkeley`

### Step by Step Guide:
This version will only connect and function properly on the newly launched Berkeley QA Network. *Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/seed-lists/berkeley_seeds.txt`.

If you are running the correct version on the correct network **BERKELEY QA NET**, `mina client status` will show:
```
Chain id:     13e6d7fd623a547387549684daee7e94ffc914a2141280e1ecafce3f5597bdf2
Git SHA-1:    144feaa08460e5e2c50c3d835e08fa8c28aa0f93
```

**Comment by deepthiskumar** [comment]
Some of things one could do to help test this release-
```
1. Sync to the network with/without empty config directory
2. Successfully able to bootstrap to the latest state after restart
3. Create snapp accounts (using the new Snapps Test Transaction CLI)
4. Update various account fields using snapp transactions (using the new Snapps Test Transaction CLI)
5. Create non-snapp accounts using snapp transactions (using the new Snapps Test Transaction CLI)
6. Invalid transactions should be rejected (Some of the errors are insufficient funds, insufficient fees, insufficient replace fees, invalid sigantures, invalid proof, duplicates, and invalid nonce )
7. Valid transactions (snapp transactions or payments or delegations) should be added to the pool and included in a block
8. Snark work is added to the pool and included in blocks
9. Community block producers are producing blocks
10. Run archive nodes and verify that transactions (snapp transactions or otherwise) are archived
```

**Comment by deepthiskumar** [comment]
Please add `berkeley-qa` label to any issues from this release that you may create

**Comment by Trivo25** [comment]
> Step by Step Guide: This version will only connect and function properly on the newly launched Berkeley QA Network. Docker only Make sure to run with --peer-list-url https://storage.googleapis.com/mina-seed-lists/berkeley_seeds.txt.


@lk86 The seed list (https://storage.googleapis.com/mina-seed-lists/berkeley_seeds.txt) seems to be empty. I am expecting this one here (https://storage.googleapis.com/seed-lists/berkeley_seeds.txt) would be the correct one?

**Reply by lk86** [reply]
fixed, thank you for catching this!
